ath79: add support for MikroTik RB951G-2HnD

kempniu opened the pull request:
MikroTik RB951G-2HnD is a wireless SOHO router that was previously supported by the ar71xx target, see commit 7a709573d7 ("ar71xx: add kernel support for the Mikrotik RB951G board").

Specifications
--------------

  - SoC: Atheros AR9344 (600 MHz)
  - RAM: 128 MB (2x 64 MB)
  - Storage: 128 MB NAND flash (various manufacturers)
  - Ethernet: Atheros AR8327 switch, 5x 10/100/1000 Mbit/s
      - 1x PoE in (port 1, 8-30 V input)
  - Wireless: Atheros AR9340 (802.11b/g/n)
  - USB: 2.0 (1A)
  - 8x LED:
      - 1x power (green, not configurable)
      - 1x user (green, not configurable)
      - 5x GE ports (green, not configurable)
      - 1x wireless (green, not configurable)
  - 1x button (restart)

Unlike on the RB951Ui-2HnD, none of the LEDs on this device seem to be GPIO-controllable, which was also the case for older OpenWRT versions that supported this board via a mach file.  The Ethernet port LEDs are controlled by the switch chip.

See https://mikrotik.com/product/RB951G-2HnD for more details.

Flashing
--------

TFTP boot initramfs image and then perform sysupgrade.  Follow common MikroTik procedures at https://openwrt.org/toh/mikrotik/common.

kempniu committed:
ath79: mikrotik: extract common bits for RB951x-2HnD devices

Mikrotik RouterBOARD 951Ui-2HnD and Mikrotik RouterBOARD RB951G-2HnD are
very similar devices.  Extract the DTS bits that are identical for these
two boards to a separate DTSI file.

Signed-off-by: Michał Kępień <openwrt@kempniu.pl>

hauke commented on target/linux/ath79/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_main.c:
```diff
@@ -1577,7 +1598,7 @@ static int ag71xx_probe(struct platform_device *pdev)
 		dev_dbg(&pdev->dev, "failed to read pll-reg property\n");
 
-	ag->pllregmap = syscon_regmap_lookup_by_phandle(np, "pll-handle");
+	ag->pllregmap = ag71xx_get_pll_regmap(np);
```
Why is `of_clk_get` returning `-EPROBE_DEFER` here?

kempniu replied:
Oh no, you caught me!  Okay, I admit that figuring out why the heck the switch does not work on this thing was frustrating enough that the prospect of having to dig into the kernel's clock API and driver binding logic got the better of me and I settled for a hack.  Now that you asked, though, I went down that rabbit hole and it turns out that the answer is simple - the `ref` DTS node (which is the clock associated with the PLL controller) did not have its `clock-frequency` property set, preventing the clock from being "initialized".

I dropped 69273c7559475034e0841e5389c34c34daf5da61 and added the `clock-frequency` property to the `ref` clock in e60ef9acf172198eaafffe87bd1021e3cbd69e11.  No more kernel hacks in this PR.  Thanks for keeping me honest.

kempniu committed:
ath79: add support for MikroTik RB951G-2HnD

MikroTik RB951G-2HnD is a wireless SOHO router that was previously
supported by the ar71xx target, see commit 7a709573d7 ("ar71xx: add
kernel support for the Mikrotik RB951G board").

Specifications
--------------

  - SoC: Atheros AR9344 (600 MHz)
  - RAM: 128 MB (2x 64 MB)
  - Storage: 128 MB NAND flash (various manufacturers)
  - Ethernet: Atheros AR8327 switch, 5x 10/100/1000 Mbit/s
      - 1x PoE in (port 1, 8-30 V input)
  - Wireless: Atheros AR9340 (802.11b/g/n)
  - USB: 2.0 (1A)
  - 8x LED:
      - 1x power (green, not configurable)
      - 1x user (green, not configurable)
      - 5x GE ports (green, not configurable)
      - 1x wireless (green, not configurable)
  - 1x button (restart)

Unlike on the RB951Ui-2HnD, none of the LEDs on this device seem to be
GPIO-controllable, which was also the case for older OpenWRT versions
that supported this board via a mach file.  The Ethernet port LEDs are
controlled by the switch chip.

See https://mikrotik.com/product/RB951G-2HnD for more details.

Flashing
--------

TFTP boot initramfs image and then perform sysupgrade.  Follow
common MikroTik procedures at https://openwrt.org/toh/mikrotik/common.

Signed-off-by: Michał Kępień <openwrt@kempniu.pl>

kempniu commented:
Thanks for merging! :heart:

Given that the 23.05 branch already has support for the 951Ui-2HnD board, is there any chance to cherry-pick this PR into the `openwrt-23.05` branch so that both 951G-2HnD and 951Ui-2HnD are supported there?
